webapp/components/MasonryGrid/MasonryGridItem.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <ds-grid-item :rowSpan="rowSpan">
+    <slot></slot>
+  </ds-grid-item>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      rowSpan: 10,
+    }
+  },
+  methods: {
+    calculateItemHeight() {
+      this.$parent.$emit('calculating-item-height')
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-20T07:03:30Z" title="Tuesday, August 20th 2019, 9:03:30 am +02:00">Aug 20, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,38 @@
+<template>
+  <ds-grid-item :rowSpan="rowSpan">
+    <slot></slot>
+  </ds-grid-item>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      rowSpan: 10,
+    }
+  },
+  methods: {
+    calculateItemHeight() {
+      this.$parent.$emit('calculating-item-height')
```


I was curious what the difference between `this.$emit` and `this.$parent.$emit` was, and found this SO post.
As far as I understand it, this is the way to tell the parent component to emit an event to its parent? so like emit an event to a grandparent component.
Is that correct @alina-beck?

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-08-20T09:12:14Z" title="Tuesday, August 20th 2019, 11:12:14 am +02:00">Aug 20, 2019</time>_

---


Yes, and the reason we have to do it here is that this component sits within a slot – I found the solution in [this comment](https://github.com/vuejs/vue/issues/4332#issuecomment-263444492) and it's working! But hard to test... 😕

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-20T21:41:27Z" title="Tuesday, August 20th 2019, 11:41:27 pm +02:00">Aug 20, 2019</time>_

---


~You could emit again in the parent component
(child)-emit->(parent)-emit->(granddaddy)~ oh no, slots (read the linked issue)